src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,226 @@
+import argparse
+import platform
+import sys
+import warnings
+from pathlib import Path
+from typing import Set
+
+from tox.execute.api import Execute
+from tox.execute.local_sub_process import LocalSubProcessExecuteInstance
+from tox.plugin import impl
+from tox.tox_env.python.api import PythonInfo
+from tox.tox_env.python.runner import PythonRun
+
+
+@impl
+def tox_add_option(parser):
+    parser.add_argument(
+        "--current-env",
+        action="store_true",
+        dest="current_env",
+        default=False,
+        help="Run tests in current environment, not creating any virtual environment",
+    )
+    parser.add_argument(
+        "--print-deps-only",
+        action="store_true",
+        dest="print_deps_only",
+        default=False,
+        help="Deprecated, equivalent to `--print-deps-to -`",
```

Comment:
If it's deprecated wouldn't switch to tox 4 be a good place to drop it?

Reply:
That's something I asked as well in the PR description.

Reply:
Agreed. Let's remove it in hooks4 and adapt the deprecation message in hooks3 to say that it will cease to exist with tox 4+.